### ISSUE TITLE
Add deepl support

### ISSUE THREAD

**Issue by werdahias** [DefunctLizard/gtktranslate#6]
Deepl has a far superior API for translating. There is a crate proving the API, unfortunately it needs a API Pro key (~7$/mo). Let me know what you think.

**Comment by grantshandy** [comment]
I looked into deeply a while ago and it looked pretty good. The only problem is the cost, I can't afford to spend a monthly fee to develop a translator I probably wouldn't use that much.

If you want to submit a PR with deepl I'd be happy to include it though!

**Comment by werdahias** [comment]
I agree that the cost is an issue; I also couldn't afford that. The problem is that the deepl-api crate requires a pro key. Maybe one could write a library that utilized the free API key. There is [dl-translate](https://crates.io/crates/dl-translate) which does a CLI translation. I will look into that
